pkg/cloudprovider/util/net.go

```diff
@@ -36,3 +37,19 @@ func CIDRToIPAndNetMask(ipv4 string) (string, string, int, error) {
 	netmask := fmt.Sprintf("%d.%d.%d.%d", ipNet.Mask[0], ipNet.Mask[1], ipNet.Mask[2], ipNet.Mask[3])
 	return ip.String(), netmask, size, nil
 }
+
+// GenerateRandMAC generates a random unicast and locally administered MAC address.
+func GenerateRandMAC() net.HardwareAddr {
+	buf := make([]byte, 6)
+	var mac net.HardwareAddr
+
+	_, err := rand.Read(buf)
+	if err != nil {
```

Comment:
I guess you want to return an error here

Reply:
yop! done. The whole `Create` is rejected if we can not compute the MAC address.